python_modules/dagster/dagster/_core/definitions/declarative_scheduling/operands/scheduled_since_condition.py

```diff
@@ -0,0 +1,142 @@
+import datetime
+from typing import Sequence, Tuple, cast
+
+from dagster._core.definitions.declarative_scheduling.serialized_objects import (
+    AssetSubsetWithMetadata,
+)
+from dagster._core.definitions.metadata.metadata_value import MetadataValue
+from dagster._serdes.serdes import whitelist_for_serdes
+
+from ..scheduling_condition import SchedulingCondition, SchedulingResult
+from ..scheduling_context import SchedulingContext
+
+_REQUEST_TIMESTAMP_METADATA_KEY = "request_timestamp"
+
+
+@whitelist_for_serdes
+class ScheduledSinceCondition(SchedulingCondition):
+    """SchedulingCondition which is true if the asset has been requested for materialization via
+    the declarative scheduling system within the given time window.
+
+    Will only detect requests which have been made since this condition was added to the asset.
+    """
+
+    lookback_days_seconds_microseconds: Tuple[int, int, int]
+
+    @property
+    def description(self) -> str:
+        return ""
+
+    @property
+    def lookback_timedelta(self) -> datetime.timedelta:
+        return datetime.timedelta(
+            days=self.lookback_days_seconds_microseconds[0],
+            seconds=self.lookback_days_seconds_microseconds[1],
+            microseconds=self.lookback_days_seconds_microseconds[2],
+        )
+
+    @staticmethod
+    def from_lookback_delta(lookback_delta: datetime.timedelta) -> "ScheduledSinceCondition":
+        lookback_days_seconds_microseconds = (
+            lookback_delta.days,
+            lookback_delta.seconds,
+            lookback_delta.microseconds,
+        )
+        return ScheduledSinceCondition(
+            lookback_days_seconds_microseconds=lookback_days_seconds_microseconds
+        )
+
+    def _get_minimum_timestamp(self, context: SchedulingContext) -> float:
+        """The minimum timestamp for a request to be considered in the lookback window."""
+        return (context.effective_dt - self.lookback_timedelta).timestamp()
+
+    def _get_new_subsets_with_metadata(
+        self, context: SchedulingContext
+    ) -> Sequence[AssetSubsetWithMetadata]:
```

Comment:
IMO we should figure out what the real name and API for `AssetSubsetWithMetadata` is going to be before doing this PR.

Reply:
Added a PR downstack that this uses